base/commands/migration/start_stages_it_test.go

```diff
@@ -0,0 +1,133 @@
+//go:build migration
+
+package migration_test
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"testing"
+	"time"
+
+	_ "github.com/hazelcast/hazelcast-commandline-client/base"
+	_ "github.com/hazelcast/hazelcast-commandline-client/base/commands"
+	"github.com/hazelcast/hazelcast-commandline-client/base/commands/migration"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+func TestMigration(t *testing.T) {
+	testCases := []struct {
+		name string
+		f    func(t *testing.T)
+	}{
+		{name: "start_Successful", f: startTest_Successful},
+		{name: "start_Failure", f: startTest_Failure},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, tc.f)
+	}
+}
+
+func startTest_Successful(t *testing.T) {
+	tcx := it.TestContext{T: t}
+	ctx := context.Background()
+	tcx.Tester(func(tcx it.TestContext) {
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go tcx.WithReset(func() {
+			defer wg.Done()
+			Must(tcx.CLC().Execute(ctx, "start", "dmt-config", "--yes"))
+		})
+		successfulRunner(tcx, ctx)
+		tcx.AssertStdoutContains(`
+Hazelcast Data Migration Tool v5.3.0
+(c) 2023 Hazelcast, Inc.
+	
+Selected data structures in the source cluster will be migrated to the target cluster.	
+
+
+ OK   [1/3] Connected to the migration cluster.
+first message
+ OK   [2/3] Started the migration.
+second message
```

Comment:
I think the order of these could change. Maybe just assert stdout contains "first message", "second message, "status report" and so on?

Reply:
I think the order will stay the same that we read from topic in order. Also asserting the order is useful.

Reply:
The main point is, this may not be easy to maintain in the future. Any new string you add in logs will make this test fail. That's why I suggested several contains instead of whole comparison. But keep it as you wish

Reply:
I think it should fail, whole comparison feels more safer.